apache/couchdb-pkg
# Add JS68 rpmspec and aarch64 RPM builds

## Overview

This pull request adds aarch64 (RHEL) support for our RPM builds. Because a good SpiderMonkey >=68 install is not available in very many RPM-based repositories, it also includes a separate `couch-js-68` rpmspec, intended to be used specifically with the `aarch64-rhel` target in the Makefile.

The incentive for this PR is that we've recently built some aarch64 RPMs for a customer at @neighbourhoodie, and @janl suggested contributing this build process to upstream. Let me know if there's something speaking against it!

There is some unanswered questions from my side:

- I have to patch out the error that `./configure` fails with when trying to build with a non-1.8.5 SpiderMonkey on aarch64. Considering that SpiderMonkey 68 and (I assume?) more recent versions work fine with aarch64, is this something that's still needed?
- I'm not sure how to integrate this into the larger Makefile-based build system. I believe that a package built on RHEL/CentOS/Amazon Linux is largely compatible with the other listed distributions, and maybe even something like Fedora, so should we add targets like `aarch64-centos-7`, `aarch64-amzn2`, and so on, and call the same routine for each of them, like we do with x64 builds already?
- I believe this should be dockerizable, but I'm not on a platform where I can easily get crossarch builds working, so I can't really test it.
- At the end of the build process, it tries to use the `PKGDIR` variable which is overridden to assume an `RPMS/arm64` subfolder, when `rpmbuild` builds in `RPMS/aarch64`. Is this something I can safely change, or are there any concerns about keeping the old arch name?
- `js68` only covers RPM builds right now, not DEB builds. Should the folder be moved somewhere else, or have a different name?

## Testing recommendations

1. Check out my fork and branch
2. Make sure you're on CentOS/RHEL/Amazon Linux/Fedora(?), and have all the build dependencies installed (the Erlang Solutions CentOS repo is very helpful in getting OTP 20 installed here)
3. Run `cd .. && git clone https://github.com/apache/couchdb`, and `make aarch64-rhel`
